client/src/components/form/form.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useContext } from 'react';
+
+const FormHeader: React.FC<React.HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
+  return (
+    <h2 {...props} >{children}</h2>
+  )
+}
+
+const FormLabel: React.FC<React.HTMLAttributes<HTMLSpanElement>> = ({ children, ...props }) => {
+  return (
+    <span {...props} >{children}</span>
+  )
+}
+
+const FormInput: React.FC<React.HTMLAttributes<HTMLElement>> = (props) => {
+  return (
+    <input {...props} />
+  )
+}
+
+const FormBody: React.FC<React.FormHTMLAttributes<HTMLFormElement>> = ({ children, ...props }) => {
+  return (
+    <form {...props} >{children}</form>
+  )
+}
+
+const FormButton: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({ children, ...props }) => {
+  return (
+    <button {...props} >{children}</button>
+  )
+}
+
+const FormClose: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({ children, ...props }) => {
+  return (
+    <FormContext.Consumer>
+      {formContext => {
+        return (
+          <button {...props} onClick={formContext.toggleShow}> {children} </button>
+        )
+      }}
+    </FormContext.Consumer>
+  )
+}
+
+const initialState = {
```

Comment:
Is this considered `state` or `context`? Since `react` has the keyword `state` reserved for `react` `state` it will be confusing in conversations regarding which `state` is being talked about.

I'm not sure if it's `initialContext` either since `context` doesn't sound like it has an `initial` value.

Looking at what the `react` doc's say it sounds like `defaultValue` is the choice.
https://reactjs.org/docs/context.html#reactcreatecontext

Reply:
this was one of the items i was having trouble with. but i'll try it with just defaultValue to see if it'll works.